pkg/solana/txm/txm.go

```diff
@@ -0,0 +1,536 @@
+package txm
+
+import (
+	"context"
+	"fmt"
+	"strings"
+	"sync"
+	"time"
+
+	solanaGo "github.com/gagliardetto/solana-go"
+	"github.com/gagliardetto/solana-go/rpc"
+	"github.com/google/uuid"
+	"github.com/pkg/errors"
+
+	"github.com/smartcontractkit/chainlink-solana/pkg/solana"
+	solanaClient "github.com/smartcontractkit/chainlink-solana/pkg/solana/client"
+	"github.com/smartcontractkit/chainlink-solana/pkg/solana/config"
+	"github.com/smartcontractkit/chainlink-solana/pkg/solana/fees"
+
+	"github.com/smartcontractkit/chainlink-solana/pkg/solana/logger"
+
+	"github.com/smartcontractkit/chainlink/core/services"
+	"github.com/smartcontractkit/chainlink/core/services/keystore"
+	"github.com/smartcontractkit/chainlink/core/utils"
```

Comment:
note these imports also show up in a few test files as well

Reply:
Services/utils - 👌 
Regarding keystore - I thought we agreed to not touch it for now. So i left it in the core (also the reason for the core dependency for the comment above). Do you want to copy it over too?

Reply:
We cannot create an import cycle, so something different needs to be done.

Reply:
There is no import cycle currently - the core branch [builds and passes tests](https://github.com/smartcontractkit/chainlink/pull/8627) successfully. Or you mean it happens after the keystore is extracted?

Reply:
Importing chainlink from chainlink-solana creates a cycle. This is not allowed.

Reply:
Well, circular dependency generates compile error. So technically since it compiles we don't have it - makes sense to me cause we aren't importing the whole core dependency tree. But I see what you mean, I'll move over the keystore too.

Reply:
You absolutely are importing the whole tree, that is the problem. You are creating a _module_ cycle, and infecting both repos with all dependencies.

Reply:
just to add to what is Jordan is saying: 

we try to avoid importing core into relays because it introduces many unneeded dependencies (in `go.mod`) that the relay does not need - you can see that in these changes where cosmos + ethereum SDKs are now in the dependency tree. this makes it incredibly difficult to update dependencies in core and relays without many PRs that are just bumping core and relay versions (especially if dependencies have breaking changes)